Generate metrics reference docs (#2164)
cilium/tetragon: docs/content/en/docs/reference/metrics.md

Reviewer: I made some comments on the output of your tool, I see two ways of tweaking stuff:
- maybe you want your output to be consistent with a previous behavior and then we need some `sed` commands to modify the output as I do with gRPC and Helm (because I don't control the tool).
- you can simply modify your tool output directly.

---
title: "Metrics"
description: >
  This reference is autogenerated from the Tetragon Prometheus metrics registry

>   using github.com/isovalent/metricstool

Reviewer: This looks a bit off since links are not handled in the description and that's referring to `isovalent` repo. Maybe we can just remove "using github.com/isovalent/metricstool", or not if you want to promote that one, but then maybe in the body?

It looks a bit weird in the section TOC as well.

<img width="1249" alt="image" src="https://github.com/cilium/tetragon/assets/11256051/6533e7f8-716a-4612-b2f7-0b2c0480960e">

Author: Yeah, this looks weird, I'll just remove it.

weight: 4
---
{{< comment >}}
This page is autogenerated via `make metrics-doc` please do not edit directly.
{{< /comment >}}
## Tetragon Health Metrics

### `tetragon_build_info`

Build information about tetragon

| label | values |
| ----- | ------ |
| `commit` | `931b70f2c9878ba985ba6b589827bea17da6ec33` |
| `go_version` | `go1.22.0` |
| `modified` | `false` |

> | `time ` | `2022-05-13T15:54:45Z` |

Reviewer: I like that it's referencing the OSS commit 😻

---

> title: "Metrics Reference"

Reviewer: I would remove this as it's already under the Reference section, like for the other ones.
```suggestion
title: "Metrics"
```

description: >

>   This reference documents Prometheus metrics exposed by Tetragon.

Reviewer: The other descriptions explain a bit how this doc is generated, with:
- This reference is generated from the Tetragon Helm chart values.
- This reference is generated from the protocol buffer specification and documents the gRPC API of Tetragon.

maybe we can add a hint as well that could interesting?

weight: 4
---

> <!-- This file is autogenerated via https://github.com/isovalent/metricstool -->

Reviewer: ```suggestion
{{< comment >}}
This page is autogenerated via https://github.com/isovalent/metricstool,
please do not edit directly.
{{< /comment >}}
```
Please use the `comment` shortcode so that Hugo removes this before generating the website, with HTML comments, they end up in the final file served by the HTTP server and are still here in prod.

Reviewer: See
<img width="1390" alt="image" src="https://github.com/cilium/tetragon/assets/11256051/078197da-ab65-4daf-9218-69004f87b8ad">

Author: I opted to drop this comment completely and document that the page is autogenerated in the frontmatter description instead. Do you think that's enough?

Author: ok, done

> # Tetragon Health Metrics

Reviewer: Could you make it a level 2 title (with all the other level 1 titles), otherwise it breaks the hierarchy and the formatting of the page 
```suggestion
## Tetragon Health Metrics
```

For example later:
<details><summary>Click to see preview</summary>
<p>
<img width="742" alt="image" src="https://github.com/cilium/tetragon/assets/11256051/f774882b-7e6e-43b6-bc17-559cd4833a0c">
</p>
</details> 

It does not expect a level 1 after the page level 1 that is already the title from the frontmatter.

> ## `tetragon_build_info`

Reviewer: Similarly, can we make every metric name a level 3 title?
```suggestion
### `tetragon_build_info`
```
